mlir/include/mlir/Dialect/Transform/IR/TransformDialect.h

```diff
@@ -315,6 +315,62 @@ class BuildOnly : public DerivedTy {
   BuildOnly() : DerivedTy(/*buildOnly=*/true) {}
 };
 
+namespace detail {
```

Comment:
Can this go to some new `Utils` header? This doesn't look like stuff that belongs to the dialect definition.

Reply:
Yep, makes sense. I realized that I only have to move the `mergeSymbolsInto` function (and related static functions). The remaining ones (for expanding paths etc) are arguably more related to the interpreter pass than this one, so I am leaving them where they are.

For now, I am calling the files `Utils.h`/`Utils.cpp`. All other files are called `TransformSomething.xxx` -- should I rename the new files to `TransformUtils.xxx` as well to follow that scheme?